lib/features/wallets/wallet_selection_page.dart

```diff
@@ -0,0 +1,170 @@
+          AsyncData(:final value) => SafeArea(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  _buildHeader(
+                    context,
+                    Loc.of(context).selectWallet,
+                    Loc.of(context).selectWalletDescription,
+                  ),
+                  Expanded(
+                    child: SizedBox(
+                      height: 24,
+                      child: _buildWalletList(context, selectedWallet, value),
+                    ),
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: Grid.side),
+                    child: FilledButton(
+                      onPressed: () async {
+                        if (selectedWallet.value?.name == 'DidPay') {
+                          Navigator.of(context).push(
+                            MaterialPageRoute(builder: (context) {
+                              return PfiVerificationPage(pfi: pfi);
+                            }),
+                          );
+                        } else {
+                          final result = await DidResolver.resolve(pfi.didUri);
+                          final widgetService = result.didDocument?.service
+                              ?.firstWhere((e) => e.type == 'IDV');
+                          final did = ref.read(didProvider);
+                          final oidcParams = await ref
+                              .read(idvServiceProvider)
+                              .getAuthRequest(
+                                  'http://${widgetService?.serviceEndpoint}',
+                                  did);
+                          LinkingService().launchWallet(oidcParams.claims.misc,
+                              selectedWallet.value?.url);
+                        }
+                      },
+                      child: Text(Loc.of(context).next),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          _ => const Center(child: CircularProgressIndicator()),
+        });
+  }
+
+  Widget _buildWalletList(
+    BuildContext context,
+    ValueNotifier<Wallet?> selectedWalletState,
+    List<Wallet> wallets,
+  ) {
+    return ListView(
+      children: wallets
+          .map(
+            (wallet) => ListTile(
+              title: Text(
+                wallet.name,
+                style: Theme.of(context).textTheme.bodyMedium?.copyWith(),
+              ),
+              leading: Container(
+                width: Grid.md,
+                height: Grid.md,
+                decoration: BoxDecoration(
+                    color: Theme.of(context).colorScheme.surface,
+                    shape: BoxShape.rectangle,
+                    borderRadius: BorderRadius.circular(Grid.xxs)),
+                child: Center(
+                  child: Text(
+                    '💳',
```

Comment:
nit: could we use a Material Icon here? maybe `Icon(Icons.wallet)`

Reply:
fixed 🫡